### PR TITLE
SPDK engine is responsible for collecting the snapshot info and actual size

### PR DESCRIPTION
**Pull request by shuo-wu**
This PR fixes part of this issue: https://github.com/longhorn/longhorn/issues/7641

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -46,9 +49,12 @@ func ProtoLvolToLvol(l *spdkrpc.Lvol) *Lvol {
...
 
 func LvolToProtoLvol(l *Lvol) *spdkrpc.Lvol {
+	if l == nil {
+		return nil
+	}
 	return &spdkrpc.Lvol{
-		Name:         l.Name,
-		Uuid:         l.UUID,
+		Name: l.Name,
+		// Uuid:         l.UUID,
```
Comment (derekbit) [inline pkg/api/types.go:57]: Can we add `UserCreated` flag? Currently, the flag is always true before finishing  https://github.com/longhorn/longhorn/issues/7578 .

Comment (shuo-wu) [reply pkg/api/types.go:57]: Do we need to handle this field in the current PR? I prefer to tackle it later.

Comment (derekbit) [reply pkg/api/types.go:57]: Sure!

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@shuo-wu Will you update instance-manager after getting merged?

**Comment by innobead** [comment]
LGTM

File: pkg/api/types.go (hunk 1 above)
Comment (innobead) [reply pkg/api/types.go:57]: @DamiaSan is working on it on SPDK part, so he should be able to help with longhorn integration here.

**Pull request merged**